README.md

```diff
@@ -1 +1,314 @@
+https://github.com/google-research/bert#pre-trained-models). You can use 
+either the 12-layer ''BERT-Base, Cased'' model or the 12-layer ''BERT-Base, 
+Uncased'' model. 
+
+### Usage Instructions
+
+**1. Data Preprocessing**
+
+The dataset we train the model on is the 
+[Microsoft Abstractive Text Compression Dataset](
+https://www.microsoft.com/en-us/download/confirmation.aspx?id=54262) (MSF dataset). 
+To preprocess the dataset, and split to train, tune, and test set, run the 
+following command
+
+```
+python preprocess_MS_dataset.py path/to/raw/data num_of_tuning num_of_testing
+```
+where we use 3,000 samples for tuning and 3,000 samples for testing in the project. 
+The preprocessed and split dataset will be saved in three tsv files named 
+train_MS_dataset, tune_MS_dataset, and tune_MS_dataset for training, tuning, and 
+testing set respectively.
+
+We also provide preprocessing scripts for three other datasets: 
+[news summary dataset](https://www.kaggle.com/sunnysai12345/news-summary), 
+[Google sentence compression dataset](
+https://github.com/google-research-datasets/sentence-compression), and 
+[reddit_tifu dataset](https://www.tensorflow.org/datasets/catalog/reddit_tifu). 
+See preprocess_news_dataset.py, preprocess_google_dataset.py, and 
+preprocess_reddit_dataset.py for code and instructions.
+
+The preprocessing script also computes basic statistics of the dataset. A 
+sample output when preprocessing the MSF dataset is
+```
+Number of samples is 26119
+Total number of excluded sample is 304
+Average word count of original sentence is 32.08 ( std: 10.79 )
+Max word count is 145
+Min word count is 7
+Average word count of shortened sentence is 22.28 ( std: 8.54 )
+Max Length is 108
+Min Length is 3
+On average, there are 1.00 sentences in each original text ( std: 0.00 )
+On average, there are 1.91 words in each shortened sentence that are not in the original sentence. ( std: 3.76 )
+The average compression ratio is 0.70 ( std: 0.19 )
+```
+
+**2. Training & Export**
+
+This streamlined process covers the steps of phrase vocabulary optimization, 
+preparing data for training, model training, and model export. The script is 
+designed for training on the Google Cloud Platform. Before running the script, 
+there are several prerequisites:
+- Create or have access to a Google Storage Bucket. Currently, the GCP bucket 
+path is set to gs://trained_models_yechen/. If you create another bucket, change 
+the path by changing the GCP_BUCKET variable in streamline_training.py.
+-  Set up a virtual machine to run the script on. Follow this [guide](
+https://docs.google.com/document/d/1oV8Swp_BDfmDHkhSkWb2wo_ZhC9jIP-Lk7kCbYvdYTM/edit#heading=h.o18hkt51hrci) 
+to set up a virtual machine on GCP.
+- If you plan to train with a Cloud TPU, follow this [guide](
+https://docs.google.com/document/d/1PlCB6DOH8LUBsN8UcgPxzds9MqPRFIPs_rht9fWqbAA/edit?usp=sharing) 
+to set up a TPU on GCP.
```

Comment:
Add a note that the TPU name and compute engine name must be the same.
Also add a note that if the zones for TPU and compute engine are different, need to set --tpu_zone with the TPU zone.

Reply:
Added both notes to the guide. Added the first note to the readme file.